projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -199,18 +209,19 @@ public event EventHandler<ConsumerTagChangedAfterRecoveryEventArgs> ConsumerTagC
         /// <summary>
         /// This event is never fired by non-recovering connections but it is a part of the <see cref="IConnection"/> interface.
         /// </summary>
-        public event EventHandler<QueueNameChangedAfterRecoveryEventArgs> QueueNameChangedAfterRecovery
+        public event AsyncEventHandler<QueueNameChangedAfterRecoveryEventArgs> QueueNameChangedAfterRecoveryAsync
         {
             add { }
             remove { }
         }
 
         internal void TakeOver(Connection other)
         {
-            _callbackExceptionWrapper.Takeover(other._callbackExceptionWrapper);
-            _connectionBlockedWrapper.Takeover(other._connectionBlockedWrapper);
-            _connectionUnblockedWrapper.Takeover(other._connectionUnblockedWrapper);
-            _connectionShutdownWrapper.Takeover(other._connectionShutdownWrapper);
+            _callbackExceptionAsyncWrapper.Takeover(other._callbackExceptionAsyncWrapper);
+            _connectionBlockedAsyncWrapper.Takeover(other._connectionBlockedAsyncWrapper);
+            _connectionUnblockedAsyncWrapper.Takeover(other._connectionUnblockedAsyncWrapper);
+            _connectionShutdownAsyncWrapper.Takeover(other._connectionShutdownAsyncWrapper);
+            // TODO Why are other wrappers not taken over?
```

Comment:
@danielmarbach my guess is that this is an oversight, I just noticed the same thing.